OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -84,7 +93,13 @@ public RallyPoint(Actor self, RallyPointInfo info)
 
 		void INotifyCreated.Created(Actor self)
 		{
-			effect = new RallyPointIndicator(self, this);
+			effect = CreateRallyPointIndicator(self);
```

Comment:
Throwing if a null effect is returned might be sensible, to avoid a more confusing error later.